tox.ini

```diff
@@ -7,14 +7,21 @@ envlist = ruff, lint, mypy, spellcheck, py3-{unit, functional}, tomllint
 minversion = 4.4
 
 [testenv]
-description = run tests (unit, unitcov, functional)
 # Use PyTorch CPU build instead of CUDA build in test envs. CUDA dependencies
 # are huge. This reduces venv from 5.7 GB to 1.5 GB.
 setenv =
     PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu
 package = wheel
 wheel_build_env = pkg
-deps = -r requirements-dev.txt
+
+[testenv:py3]
+description = run tests (unit, unitcov, functional)
+basepython = python3.11
```

Comment:
Doesn't this mean that we never test on python 3.10 now? The test.yml worklflow includes python 3.10. But then this overrides that and tests with 3.11?

Reply:
tox's rules for inheritance and Python version mapping are not very intuitive. It's one of the reasons I personally prefer nox over tox.

Tox default:
- `[toxenv:py3]`, `[toxenv:py310]`, and `[toxenv:py311]` are independent and inherit settings from `[toxenv]`.
- `[toxenv:py3]` uses `python3` binary as default base Python. This can point to any Python 3 version. On my system, it is currently Python 3.12
- `[toxenv:py310]` uses `python3.10` binary as default base Python
- `[toxenv:py311]` uses `python3.11` binary as default base Python

To my surprise, ``[toxenv:py3-unit]` does not inherit settings from `[toxenv:py3]`.

My changes modifies `[toxenv:py3-unit]` and `[toxenv:py3-function]` to use `python3.11` instead of `python3`, so `tox -e py3-unit` works when `python3` points to `python3.12`.

Our CI already maps GH envs to tox envs with fully qualified Python version, so we properly test with 3.10 and 3.11.

```
[gh]
python =
    3.11 = py311-{unitcov, functional}
    3.10 = py310-{unitcov, functional}
```

Reply:
OK, but the unit tests are not running in this configuration. Look at the output from py3-unit. It does not call pytest. It just sets up pytest for execution but does not run it.

Reply:
Mpf, I got fooled by tox's non-intuitive inheritance again. Fixing now...